subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -383,6 +422,218 @@ static int cmd_wifi_stats(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_wifi_ps(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_ps_params params = { 0 };
+
+	context.sh = sh;
+
+	if (argc > 2) {
+		shell_fprintf(sh, SHELL_WARNING, "Invalid number of arguments\n");
+		return -ENOEXEC;
+	}
+
+	if (argc == 1) {
+		struct wifi_ps_config config = { 0 };
+
+		if (net_mgmt(NET_REQUEST_WIFI_PS_CONFIG, iface,
+				&config, sizeof(config))) {
+			shell_fprintf(sh, SHELL_WARNING, "Failed to get PS config\n");
+			return -ENOEXEC;
+		}
+
+		shell_fprintf(sh, SHELL_NORMAL, "PS status: %s\n",
+				wifi_ps_mode_txt(config.enabled));
+		if (config.enabled) {
+			shell_fprintf(sh, SHELL_NORMAL, "PS mode: %s\n",
+					wifi_ps_mode_txt(config.mode));
+		}
+
+		if (config.num_twt_flows == 0) {
+			shell_fprintf(sh, SHELL_NORMAL, "No TWT flows\n");
+		} else {
+			for (int i = 0; i < config.num_twt_flows; i++) {
+				shell_fprintf(sh, SHELL_NORMAL, "TWT Dialog token: %d\n",
+					config.twt_flows[i].dialog_token);
+				shell_fprintf(sh, SHELL_NORMAL, "TWT flow ID: %d\n",
+					config.twt_flows[i].flow_id);
+				shell_fprintf(sh, SHELL_NORMAL, "TWT negotiation type: %s\n",
+					wifi_twt_negotiation_type_txt(
+					config.twt_flows[i].negotiation_type));
+				shell_fprintf(sh, SHELL_NORMAL, "TWT responder: %s\n",
+					config.twt_flows[i].responder ? "true" : "false");
+				shell_fprintf(sh, SHELL_NORMAL, "TWT implicit: %s\n",
+					config.twt_flows[i].implicit ? "true" : "false");
+				shell_fprintf(sh, SHELL_NORMAL, "TWT trigger: %s\n",
+					config.twt_flows[i].trigger ? "true" : "false");
+				shell_fprintf(sh, SHELL_NORMAL, "TWT announce: %s\n",
+					config.twt_flows[i].announce ? "true" : "false");
+				shell_fprintf(sh, SHELL_NORMAL, "TWT wake interval: %d ms\n",
+					config.twt_flows[i].twt_wake_interval_ms);
+				shell_fprintf(sh, SHELL_NORMAL, "TWT interval: %d ms\n",
+					config.twt_flows[i].twt_interval_ms);
+				shell_fprintf(sh, SHELL_NORMAL, "========================\n");
+			}
+		}
```

Comment:
Shouldn't there be a return here?

Reply:
Why? The intention is to print info about all flows

Reply:
I meant if there is no return here, then the code will fall through and try to execute further instructions which is not intended.

Reply:
Ah sorry, didn't see the diff below, will fix, thanks.